src/current/v24.3/ldap.md

```diff
@@ -0,0 +1,206 @@
+
+1. During login, CockroachDB fetches the user's group memberships from LDAP.
+2. Groups are mapped to CockroachDB roles using the group's Common Name (CN).
+3. The user is automatically granted membership to corresponding roles.
+4. Any existing role memberships not matched to LDAP groups are revoked.
+
+### Configuration
+
+#### Step 1: Enable LDAP Authorization
+
+Add the `ldapgrouplistfilter` parameter to your HBA configuration:
+
+~~~ sql
+SET CLUSTER SETTING server.host_based_authentication.configuration = '
+host    all    all    all    ldap    ldapserver=ldap.example.com 
+    ldapport=636 
+    "ldapbasedn=ou=users,dc=example,dc=com" 
+    "ldapbinddn=cn=readonly,dc=example,dc=com" 
+    ldapbindpasswd=readonly_password 
+    ldapsearchattribute=uid 
+    "ldapsearchfilter=(memberof=cn=cockroachdb_users,ou=groups,dc=example,dc=com)"
+    "ldapgrouplistfilter=(objectClass=groupOfNames)"
+host    all    root    0.0.0.0/0    password';
+~~~
+
+The `ldapgrouplistfilter` parameter defines which LDAP objects should be considered as groups.
+
+Q: Get more specific/realistic example from Eng?
+
+#### Step 2: Create Matching Roles
+
+Create CockroachDB roles that match your LDAP group names and grant appropriate privileges to each role.
+
+For an LDAP group `cn=db_admins,ou=groups,dc=example,dc=com`:
+
+~~~ sql
+CREATE ROLE db_admins;
+GRANT ALL ON DATABASE app TO db_admins;
+~~~
+
+### Group Search Template
+
+You can configure how LDAP groups are searched using the `server.ldap.group_search_template` setting:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap.group_search_template = 'ou=groups,dc=example,dc=com??sub?(&(objectClass=groupOfNames)(member={USER_DN}))';
+~~~
+
+The template follows RFC4515/4516 format:
+
+- Base DN: Where to start the search
+- Scope: How deep to search (`sub` for subtree)
+- Filter: Criteria for matching groups
+- `{USER_DN}`: Placeholder for the authenticated user's DN
+
+### Troubleshooting
+
+Enable authentication logging to troubleshoot LDAP issues:
+
+~~~ sql
+SET CLUSTER SETTING server.auth_log.sql_sessions.enabled = true;
+~~~
+
+Potential issues may pertain to:
+
+- Network connectivity to the LDAP server.
+- Incorrect bind DN or password.
+- Search filter not matching the intended users.
+- TLS certificates.
+- Missing or mismatched role names.
+
+### Security Considerations
+
+1. Always keep a backup authentication method (like password) for administrative users
+2. Use LDAPS (LDAP over TLS) in production environments
+3. Use a restricted service account for directory searches
+4. Regularly audit LDAP group memberships
+5. Monitor authentication logs for unusual patterns
+
+### Database Console Integration
+
+LDAP authentication is also available for the DB Console. Users can log in using their LDAP credentials if configured in the HBA configuration.
+
+Q: Do you have more detail that we should include on this?
```

Comment:
LDAP auth for db console works exactly the same way as cluster sso for ldap. We provide sql username and ldap_password as creds. We validate that HBA config for the sql user is pointing to ldap as auth method and then proceed with authN if user, incoming ip address and database match HBA entry. If the authN fails, we revert to using password authN and use the given creds to check if it is matching the stored password for user in DB. If LDAP auth method is not set, we still continue with password authN as usual. No authZ happens incase of db console login.

Reply:
We fall back to username/password auth in any of the 4 cases:
1. if the user is not found in the directory
2. if LDAP credentials are not valid
3. if HBA config settings are not valid
4. if no HBA rule matches the ldap method for the authenticating user
As per my understanding there are the permissions which are useful for DB console management: https://www.cockroachlabs.com/docs/v23.2/ui-overview#role-based-security
`VIEWCLUSTERMETADATA`, `VIEWJOB`, etc. These will need to be directly assigned to user and not inherited via authZ as that was not previously the case and unsure if that needs to be supported as well.

Reply:
This was currently not planned and previously authorization did not happen on DB console. The user could potentially do a cluster sso first which assigns all the relevant groups to the user with privileges. On accessing DB console these permissions are persisted and we don't do any reassignment or dropping of group roles for user. cc: @dikshant

Reply:
@biplav I think we should note this gap as we consider GA for this later.